core/pkg/server/connection.go

```diff
@@ -244,6 +245,12 @@ func (nc *Connection) handleServerRequest() {
 func (nc *Connection) handleInformInit(msg *service.ServerInformInitRequest) {
 	settings := settings.From(msg.GetSettings())
 
+	// cancel context after settings.GetInitTimeout()
+	timeout := time.Duration(settings.GetInitTimeout() * float64(time.Second))
+	ctx, cancel := context.WithTimeout(nc.ctx, timeout)
+	// nc.ctx = ctx
```

Comment:
i think this would only work for init, but we probably need to wire up the cancel that comes from the mailbox logic :/